test/components/componentTests.ts

```diff
@@ -246,333 +344,338 @@ describe("Component behavior", () => {
-      assert.strictEqual(origin.y, SVG_HEIGHT - cHeight, "returns correct value (yAlign bottom)");
+      assert.isFalse((<Node> svg.node()).hasChildNodes(), "removed from svg");
+      assert.isTrue((<Node> svg2.node()).hasChildNodes(), "anchored onto second svg");
+      assert.strictEqual(c.height(), svgHeight2, "occupies entire space of second svg");
 
+      c.destroy();
+      svg2.remove();
       svg.remove();
     });
+  });
 
-    it("originToSVG() (nested)", () => {
-      TestMethods.fixComponentSize(c, cWidth, cHeight);
-      let group = new Plottable.Components.Group([c]);
-      group.renderTo(svg);
-
-      let groupWidth = group.width();
-      let groupHeight = group.height();
-
-      c.xAlignment("left").yAlignment("top");
-      let origin = c.originToSVG();
-      assert.strictEqual(origin.x, 0, "returns correct value (xAlign left)");
-      assert.strictEqual(origin.y, 0, "returns correct value (yAlign top)");
-
-      c.xAlignment("center").yAlignment("center");
-      origin = c.originToSVG();
-      assert.strictEqual(origin.x, (groupWidth - cWidth) / 2, "returns correct value (xAlign center)");
-      assert.strictEqual(origin.y, (groupHeight - cHeight) / 2, "returns correct value (yAlign center)");
+  it("does not allow the origin object to be modified", () => {
+    c.renderTo(svg);
+    let receivedOrigin = c.origin();
+    let delta = 10;
+    receivedOrigin.x += delta;
+    receivedOrigin.y += delta;
+    assert.notStrictEqual(receivedOrigin.x, c.origin().x, "can be modified without affecting origin (x)");
+    assert.notStrictEqual(receivedOrigin.y, c.origin().y, "can be modified without affecting origin (y)");
+    c.destroy();
+    svg.remove();
+  });
 
-      c.xAlignment("right").yAlignment("bottom");
-      origin = c.originToSVG();
-      assert.strictEqual(origin.x, groupWidth - cWidth, "returns correct value (xAlign right)");
-      assert.strictEqual(origin.y, groupHeight - cHeight, "returns correct value (yAlign bottom)");
+  describe("calculating the origin in relation to the svg", () => {
+    it("returns origin without a parent", () => {
+      assert.deepEqual(c.originToSVG(), c.origin(), "same as origin with no parent");
+      c.destroy();
+      svg.remove();
+    });
 
+    it("is offset by the parent's origin", () => {
+      let parent = new Plottable.Components.Group([c]);
+      parent.anchor(svg);
+      c.anchor(svg);
+      parent.computeLayout({x: SVG_WIDTH / 4, y: SVG_HEIGHT / 4}, SVG_WIDTH / 2, SVG_HEIGHT / 2);
+      c.computeLayout({x: SVG_WIDTH / 4, y: SVG_HEIGHT / 4}, SVG_WIDTH / 4, SVG_HEIGHT / 4);
+      let originToSvg = {
+        x: parent.origin().x + c.origin().x,
+        y: parent.origin().y + c.origin().y
+      };
+      assert.deepEqual(c.originToSVG(), originToSvg, "origin offsetted by parents");
+      parent.destroy();
+      c.destroy();
       svg.remove();
     });
   });
+
+  it("allows for recomputing the layout and rendering", () => {
```

Comment:
1. so that we don't need beforeEach at the most outer level (but this is related to the nesting beforeEach discussion above)
2. it seems to me that it's a convention that we put everything in describe blocks in other test files. if not, it's okay, but at least put all the stand alone tests before describe blocks so it's easy to read and find